src/metax_api/services/common_service.py

```diff
@@ -469,3 +470,28 @@ def set_if_modified_since_filter(cls, request, filter_obj):
                 filter_obj['q_filters'].append(flter)
             else:
                 filter_obj['q_filters'] = [flter]
+
+    @staticmethod
+    def identifiers_to_ids(identifiers, params=None):
+        """
+        In case identifiers are identifiers (strings), which they probably are in real use,
+        do a query to get a list of pk's instead, since they will be used quite a few times.
+        """
+        if not isinstance(identifiers, list):
+            raise Http400('Received identifiers is not a list')
+        elif not identifiers:
+            _logger.info('Received empty list of identifiers. Aborting')
+            raise Http400('Received empty list of identifiers')
+        elif all(isinstance(x, int) for x in identifiers):
+            return identifiers
+
+        if params in ['files', 'noparams']:
+            from metax_api.models import File
+            identifiers = [ id for id in File.objects.filter(identifier__in=identifiers).values_list('id', flat=True) ]
+        else:
+            from metax_api.models import CatalogRecord as cr
+            identifiers = [ id for id in cr.objects.filter(identifier__in=identifiers).values_list('id', flat=True) ]
+
+        if not identifiers:
+            raise Http404
+        return identifiers
```

Comment:
Brought this method to common methods to be used more widely, but not sure if it's OK to do imports for models from common class. (Imports are located here temporarily)